src/functional/iterator/transforms/simple_inline_heuristic.py

```diff
@@ -0,0 +1,36 @@
+from collections.abc import Callable
+
+from eve import NodeVisitor
+from functional.iterator import ir
+
+
+class AnalyzeLifts(NodeVisitor):
```

Comment:
Please document the strategy. This just prevents inlining scans?

Reply:
Also document in the right place (in the global_tmps pass) that this is required for scans and why.

Reply:
It prevents inlining scans and inling of the first lifted functions within a scan. Otherwise we would get scan-only computations. It’s only strictly needed by gtfn, I added comments (to the `heuristic` function).